### PR TITLE
GE-158  get_latest_files date fix

### PR DESCRIPTION
**Pull request by crux-matthijsk**
GE-158 get_latest_files date fix

### CODE REVIEW

**Comment by stu-speed** [comment]
Put description in PR title and use same description when you squash and merge

**Review by stu-speed — CHANGES_REQUESTED** [review]

File: crux/_utils.py [hunk 1]
```diff
@@ -50,7 +50,10 @@ def url_builder(url_base, url_prefix, url_path_list):
     Returns:
         str: Valid Percent Encoded URL.
     """
-    if url_prefix:
+
+    if 'v1' in url_prefix:
+        valid_url = [url_base, url_prefix]
```
Comment (stu-speed) [inline crux/_utils.py:55]: is this needed?  should not be different than v2 case

File: crux/models/dataset.py [hunk 2]
```diff
@@ -1334,13 +1334,13 @@ def get_files_range(
         delivery_status = "DELIVERY_SUCCEEDED" if delivery_status is None else delivery_status
         use_cache = True if use_cache is None else use_cache
         params = {
-            "start_date": stdt,
+            "start_date": datetime.isoformat(stdt),
             "end_date": None if enddt is None else enddt + timedelta(days=3),
```
Comment (stu-speed) [inline crux/models/dataset.py:1338]: enddt+t should also be converted to isoformat

File: crux/_config.py [hunk 3]
```diff
@@ -80,6 +81,12 @@ def __init__(  # pylint: disable=too-many-branches
             self.api_prefix_v2 = api_prefix_v2
         log.trace("Setting API prefix-v2 to %s", self.api_prefix_v2)
 
+        if api_prefix_v1 is None:
+            self.api_prefix_v1 = os.environ.get("CRUX_API_PREFIX_V1", "v1/client")
+        else:
+            self.api_prefix_v1 = api_prefix_v1
+
+        log.trace("Setting API prefix-v2 to %s", self.api_prefix_v2)
```
Comment (stu-speed) [inline crux/_config.py:89]: v1

File: crux/models/dataset.py (hunk 2 above)
Comment (crux-matthijsk) [reply crux/models/dataset.py:1338]: The tests I did didn't have end_date. I will have to change that and test it another time

Comment (crux-matthijsk) [reply crux/models/dataset.py:1338]: changed that

File: crux/_utils.py (hunk 1 above)
Comment (stu-speed) [reply crux/_utils.py:55]: If it is an issue then why does v2 not have a problem

File: crux/models/dataset.py [hunk 4]
```diff
@@ -1334,13 +1334,13 @@ def get_files_range(
         delivery_status = "DELIVERY_SUCCEEDED" if delivery_status is None else delivery_status
         use_cache = True if use_cache is None else use_cache
         params = {
-            "start_date": stdt,
-            "end_date": None if enddt is None else enddt + timedelta(days=3),
+            "start_date": datetime.isoformat(stdt),
+            "end_date": None if enddt is None else datetime.isoformat(enddt) + timedelta(days=3),
```
Comment (stu-speed) [inline crux/models/dataset.py:1338]: Needs to be datetime.isoformat(enddt+ timedelta(days=3))

**Comment by stu-speed** [comment]
GE-158 get_latest_files date fix

**Review by stu-speed — APPROVED** [review]

File: crux/_config.py [hunk 5]
```diff
@@ -80,6 +81,12 @@ def __init__(  # pylint: disable=too-many-branches
             self.api_prefix_v2 = api_prefix_v2
         log.trace("Setting API prefix-v2 to %s", self.api_prefix_v2)
 
+        if api_prefix_v1 is None:
+            self.api_prefix_v1 = os.environ.get("CRUX_API_PREFIX_V1", "v1/client")
```
Comment (spmistry) [inline crux/_config.py:85]: for v2 the default prefix is just v2
should that be made consistent to `v2/client`
[swagger ref](https://api.cruxinformatics.com/v2/client/swagger#/)

File: crux/_utils.py (hunk 1 above)
Comment (spmistry) [reply crux/_utils.py:55]: quote method is just to encode a url containing special chars
this shouldnt need any special handling for v1 case

File: crux/models/dataset.py [hunk 6]
```diff
@@ -1334,13 +1334,13 @@ def get_files_range(
...
-            "start_date": stdt,
-            "end_date": None if enddt is None else enddt + timedelta(days=3),
+            "start_date": datetime.isoformat(stdt),
+            "end_date": None if enddt is None else datetime.isoformat(enddt + timedelta(days=3)),
             "delivery_status": delivery_status,
             "use_cache": use_cache,
         }
         response = self.connection.api_call(
-            "GET", ["deliveries", self.id, "ids"], headers=headers, params=params
+            "GET", ["v1", "deliveries", self.id, "ids"], headers=headers, params=params
```
Comment (spmistry) [inline crux/models/dataset.py:1343]: this should be `[v2, client, deliveries, self.id, ids]`
to keep it consistet with existing structure for v2

File: crux/models/dataset.py [hunk 7]
```diff
@@ -1356,7 +1356,7 @@ def get_files_range(
                 raise ValueError("Value of delivery_id is invalid")
             params = {"delivery_resource_format": file_format}
             response = self.connection.api_call(
-                "GET", ["deliveries", self.id, delivery_id, "data"], params=params
+                "GET", ["v1", "deliveries", self.id, delivery_id, "data"], params=params
```
Comment (spmistry) [inline crux/models/dataset.py:1359]: same here. use  `[v1, client, deliveries, self.id, delivery_id, data]`

File: crux/apis.py [hunk 8]
```diff
@@ -59,7 +59,7 @@ def whoami(self):
             {"accept": "application/json"}
         )  # type: Optional[MutableMapping[Text, Text]]
         return self.api_client.api_call(
-            "GET", ["identities", "whoami"], model=Identity, headers=headers
+            "GET", ["v1", "identities", "profile"], model=Identity, headers=headers
```
Comment (spmistry) [inline crux/apis.py:62]: use `v1, client` in both endpoints in this file

**Review by spmistry — CHANGES_REQUESTED** [review]
Pls update the inconsistent behavior across v1 and v2 calls structuring.

File: crux/apis.py (hunk 8 above)
Comment (stu-speed) [reply crux/apis.py:62]: @spmistry  the purpose of the prefix is to expose what the library has access to.  It should only have access to v1/client in the future but during the transition, the client has access to v2 and plat-api prefixes.  The code will be consistent once the transition is complete.

**Comment by stu-speed** [comment]
@spmistry  v1 and v2 are already inconsistent as there is no v2/client.  That is why the extra code is there.  v2 does not have full support for all api calls and is relying on v1/client as documented in swagger.

**Review by spmistry — APPROVED** [review]

**Pull request merged**